engines/v8/src/wasi.cc

```diff
@@ -0,0 +1,48 @@
+#include "wasi.hh"
+#include <cstdlib>
+#include <iostream>
+
+// WebAssembly type: (func (param i32))
+auto proc_exit(const wasm::Val args[], wasm::Val results[])
+    -> wasm::own<wasm::Trap> {
+  std::cerr << "proc_exit" << std::endl;
+  // exit(args[0].i32()); // TODO this cannot actually exit here; should trap?
```

Comment:
Yes, WASI traps when you call `proc_exit`, and saves the exit code in some side state so the embedder can differentiate between an exit and another kind of trap.